### PR TITLE
reset global ttp value

### PR DESCRIPTION
**Pull request by pondrejk**
The time to pickup value for pull provider rex jobs was set globally by one of the tests but not wasn't reset back to default, which caused interference with other pull-provider tests

### CODE REVIEW

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup

File: tests/foreman/api/test_remoteexecution.py [hunk 1]
```diff
@@ -190,6 +190,7 @@ def test_negative_time_to_pickup(
     global_ttp = module_target_sat.api.Setting().search(
         query={'search': 'name="remote_execution_time_to_pickup"'}
     )[0]
+    default_global_ttp = global_ttp.value
```
Comment (jyejare) [inline tests/foreman/api/test_remoteexecution.py:193]: For restoring the setting, you should use `setting_update` fixture!

**Review by jyejare — CHANGES_REQUESTED** [review]
Use `setting_update` Fixture!

File: tests/foreman/api/test_remoteexecution.py (hunk 1 above)
Comment (pondrejk) [reply tests/foreman/api/test_remoteexecution.py:193]: I can't do it in a fixture as I need it to be default for the first part of the test. Anyway, I decided to have this part separated to a test of its own which should solve the problem

Comment (ogajduse) [reply tests/foreman/api/test_remoteexecution.py:193]: He can not use this fixture as he does not want the setting to be applied at the time when the first Job Invocation is started. Therefore the `setting_update fixture can not be used in this test.

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup

File: tests/foreman/api/test_remoteexecution.py (hunk 1 above)
Comment (jyejare) [reply tests/foreman/api/test_remoteexecution.py:193]: This test is so big and I didnt realize that the update setting is not needed in first job as u said!

**Review by jyejare — APPROVED** [review]

File: tests/foreman/api/test_remoteexecution.py [hunk 2]
```diff
@@ -214,3 +215,5 @@ def test_negative_time_to_pickup(
         query={'search': f'resource_type = JobInvocation and resource_id = {job["id"]}'}
     )
     assert 'The job was not picked up in time' in result[0].humanized['output']
+    global_ttp.value = default_global_ttp
+    global_ttp.update(['value'])
```
Comment (jameerpathan111) [inline tests/foreman/api/test_remoteexecution.py:219]: @pondrejk This still doesn’t completely resolve the original issue. The test can fail before global ttp value is reset.

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup[rhel8]

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup[rhel8]

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup

**Comment by pondrejk** [comment]
prt 3505 passed with:

```
pytest tests/foreman/api/test_remoteexecution.py -k test_negative_time_to_pickup --external-logging --junit-xml=sat-results.xml -o junit_suite_name=sat-result
08:16:47  ============================= test session starts ==============================
08:16:47  platform linux -- Python 3.11.4, pytest-7.4.0, pluggy-1.2.0
08:16:49  Mandatory Requirements are up to date.
08:16:49  Optional Requirements Mismatch: pre-commit==3.3.3 manage==0.1.15 pytest-cov==4.1.0 sphinx==7.2.0 flake8==6.1.0 redis==5.0.0 sphinx-autoapi==2.1.1
08:16:49  To update mismatched requirements, run the pytest command with '--upgrade-required-reqs' OR '--upgrade-all-reqs' option.
08:16:49  shared_function enabled - OFF - scope:  - storage: file
08:16:49  rootdir: /opt/app-root/src/robottelo
08:16:49  configfile: pyproject.toml
08:16:49  plugins: ibutsu-2.2.4, mock-3.11.1, reportportal-5.2.1, services-2.2.1, xdist-3.3.1
08:16:52  collected 2 items / 1 deselected / 1 selected
08:16:52  
08:52:35  tests/foreman/api/test_remoteexecution.py .                              [100%]
```

**Review by ogajduse — APPROVED** [review]

**Pull request merged**